src/ralph/backends/http/async_lrs.py

```diff
@@ -0,0 +1,361 @@
+
+        Args:
+            base_url (AnyHttpUrl): LRS server URL.
+            username (str): Basic auth username for LRS authentication.
+            password (str): Basic auth password for LRS authentication.
+            headers (dict): Headers defined for the LRS server connection.
+            status_endpoint (str): Endpoint used to check server status.
+            statements_endpoint (str): Default endpoint for LRS statements resource.
+        """
+        self.base_url = parse_obj_as(AnyHttpUrl, base_url)
+        self.auth = (username, password)
+        self.headers = headers
+        self.status_endpoint = status_endpoint
+        self.statements_endpoint = statements_endpoint
+
+    async def status(self):
+        """HTTP backend check for server status."""
+        status_url = urljoin(self.base_url, self.status_endpoint)
+
+        try:
+            async with AsyncClient() as client:
+                response = await client.get(status_url)
+            response.raise_for_status()
+        except RequestError:
+            msg = "Unable to request the server"
+            logger.error(msg)
+            return HTTPBackendStatus.AWAY
+        except HTTPStatusError:
+            msg = "Response raised an HTTP status of 4xx or 5xx"
+            logger.error(msg)
+            return HTTPBackendStatus.ERROR
+
+        return HTTPBackendStatus.OK
+
+    async def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """Raise error for unsupported `list` method."""
+        msg = "LRS HTTP backend does not support `list` method, cannot list from " "%s"
+
+        logger.error(msg, target)
+        raise NotImplementedError(msg % target)
+
+    async def _fetch_statements(self, target, raw_output, query_params: dict):
+        """Fetch statements from a LRS. Used in `read`."""
+        async with AsyncClient(
+            auth=self.auth, headers=self.headers.dict(by_alias=True)
+        ) as client:
+            while True:
+                response = await client.get(target, params=query_params)
+                response.raise_for_status()
+
+                statements_response = StatementResponse.parse_obj(response.json())
+                statements = statements_response.statements
+                statements = (
+                    [statements] if not isinstance(statements, list) else statements
+                )
+                if raw_output:
+                    for statement in statements:
+                        yield bytes(json.dumps(statement), encoding="utf-8")
+                else:
+                    for statement in statements:
+                        yield statement
+
+                if not statements_response.more:
+                    break
+
+                query_params.update(parse_qs(urlparse(statements_response.more).query))
+
+    async def _greedy_fetch_statements(self, target, raw_output, query_params):
```

Comment:
nit: I like having private methods at the end of the class, after public methods, for clarity. WDYT?

Reply:
I was not aware. Just changed it !